tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -364,11 +369,15 @@ struct SwiftASTManager::Implementation {
                            "sourcekit.swift.ASTBuilding" };
 
   ASTProducerRef getASTProducer(SwiftInvocationRef InvokRef);
-  FileContent getFileContent(StringRef FilePath, bool IsPrimary,
-                             std::string &Error);
-  BufferStamp getBufferStamp(StringRef FilePath);
-  std::unique_ptr<llvm::MemoryBuffer> getMemoryBuffer(StringRef Filename,
-                                                      std::string &Error);
+  FileContent
+  getFileContent(StringRef FilePath, bool IsPrimary, std::string &Error,
+                 llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem);
+  BufferStamp
+  getBufferStamp(StringRef FilePath,
+                 llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem);
+  std::unique_ptr<llvm::MemoryBuffer> getMemoryBuffer(
+      StringRef Filename, std::string &Error,
+      llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem = nullptr);
```

Comment:
It looks somewhat inconsistent that only one of these methods has a default value for the newly introduced `FileSystem` parameter.

Reply:
will be addressed by the small refactoring